### PR TITLE
Collect all comments

### LINKED ISSUE #632: Some comments are missing in swc_ecma_parser 0.18.0
**Describe the bug**

It was strange, but I only noticed this after clearing out my cargo.lock file, so I think this might be a problem with a downstream dependency.

**Input code**

```ts
if (true) { /* 1 */ // 2
    true;
    true;
}
```

**Expected behavior**

Both comments.

**Actual behaviour**

Only one comment:

```
Leading: {}
Trailing: {BytePos(11): [Comment { kind: Line, span: Span { lo: BytePos(20), hi: BytePos(24), ctxt: #0 }, text: " 2" }]}
```

Using this code on a `swc_common::comments::Comments`:

```rust
let (leading_comments, trailing_comments) = comments.take_all();

println!("Leading: {:?}", leading_comments);
println!("Trailing: {:?}", trailing_comments);
```

Full file is [here](https://github.com/dsherret/dprint/blob/8983ce321a24bebcae9b2fd3a0600c049644c2d2/packages/rust-dprint-plugin-typescript/src/parse_swc_ast.rs)

**Version**

0.18.0 of swc_ecma_parser
0.5.0 of swc_common
0.16.0 of swc_ecma_ast

### PR DESCRIPTION
**Pull request by kdy1**
Closes #632.

### CODE REVIEW

File: common/src/comments.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ impl Default for Comments {
 
 impl Comments {
     pub fn add_leading(&self, pos: BytePos, cmt: Vec<Comment>) {
-        self.leading.insert(pos, cmt);
+        self.trailing.entry(pos).or_default().extend(cmt);
```
Comment (dsherret) [inline common/src/comments.rs]: @kdy1 should this be `leading`?

Comment (kdy1) [reply common/src/comments.rs]: Nice catch! :P

**Comment by kdy1** [comment]
bors r+

I tested locally and it passed.

**Comment by kdy1** [comment]
Tests were successful locally before `cargo update`.

I did `cargo update`.
```sh
    Updating crates.io index
    Updating aho-corasick v0.7.6 -> v0.7.8
    Updating atty v0.2.13 -> v0.2.14
      Adding autocfg v1.0.0
    Updating const-random v0.1.6 -> v0.1.8
    Updating const-random-macro v0.1.6 -> v0.1.8
    Updating dashmap v3.4.0 -> v3.4.2
    Updating getrandom v0.1.13 -> v0.1.14
    Updating indexmap v1.3.0 -> v1.3.2
    Updating itoa v0.4.4 -> v0.4.5
    Updating memchr v2.2.1 -> v2.3.0
    Updating num-bigint v0.2.4 -> v0.2.6
    Updating num-integer v0.1.41 -> v0.1.42
    Updating num-traits v0.2.10 -> v0.2.11
    Updating proc-macro2 v1.0.7 -> v1.0.8
    Removing qadapt-spin v1.0.1
    Updating rand v0.7.2 -> v0.7.3
    Updating regex v1.3.1 -> v1.3.4
    Updating regex-syntax v0.6.12 -> v0.6.14
    Updating same-file v1.0.5 -> v1.0.6
    Updating serde_json v1.0.44 -> v1.0.46
    Updating smallvec v1.1.0 -> v1.2.0
    Updating syn v1.0.13 -> v1.0.14
    Updating termcolor v1.0.5 -> v1.1.0
    Updating thread_local v0.3.6 -> v1.0.1
    Updating unicode-normalization v0.1.11 -> v0.1.12
    Updating wasi v0.7.0 -> v0.9.0+wasi-snapshot-preview1
    Updating winapi-util v0.1.2 -> v0.1.3
    Removing wincolor v1.0.2

```

**Comment by kdy1** [comment]
bors r-

I'll merge it manually

**Pull request merged**